### PR TITLE
Fixes #245 by removing deferral of rendering until document loaded

### LINKED ISSUE #245: Distribution broken when using native custom elements and parser generated nodes
Recently merged https://github.com/webcomponents/shadydom/pull/222 introduced an issue with re-distribution when creating shadowRoots during document load. 

Support for this was added to interoperate with native custom elements which can create shadowRoot's while the document is parsing. Fixing it looks complicated and instead this change should be rolled back.

Instead a flag can be added to specifically defer native custom element's connection state processing until document load has occurred. Since this may have negative performance impact, the flag should only be used during testing.

### PR DESCRIPTION
**Pull request by sorvell**
* In general, shadowRoot's should not be created on elements in the DOM until the parser has completely parsed them.
* However, a flag `ShadyDOM.deferNativeCustomElementsConnection` can be turned on to prevent custom elements from connecting until the document has loaded. If shadowRoot's are created during connectedCallback, this can enable native custom elements to create shadowRoots during document load (because the work is deferred until after load). This is intended for testing and should be avoided in production code since it may have negative perf impact.

### CODE REVIEW

File: src/attach-shadow.js [hunk 1]
```diff
@@ -30,7 +29,7 @@ const SHADYROOT_NAME = 'ShadyRoot';
 
 const MODE_CLOSED = 'closed';
 
-let isRendering = false;
+let isRendering = utils.settings['deferNativeCustomElementsConnection'] && document.readyState === 'loading';
```
Comment (kevinpschaaf) [inline src/attach-shadow.js]: Consider name change: `deferConnectionCallbacks`

File: src/attach-shadow.js [hunk 2]
```diff
@@ -153,7 +137,8 @@ class ShadyRoot {
       for (let i=0, l=c$.length; i < l; i++) {
         const child = c$[i];
         const data = shadyDataForNode(child);
-        if (parentNode(child) === this.host && !data.assignedSlot) {
+        if (parentNode(child) === this.host &&
+            (child.localName === 'slot' || !data.assignedSlot)) {
```
Comment (kevinpschaaf) [inline src/attach-shadow.js]: TODO: why is slot special handling required here?

Comment (sorvell) [reply src/attach-shadow.js]: Elements are not composed into shadowRoot hosts. Therefore all non-rendering children must be manually removed at first render. This includes unassigned nodes and slots.

**Comment by azakus** [comment]
Rerunning build with webcomponents/template#46

File: src/attach-shadow.js (hunk 1 above)
Comment (sorvell) [reply src/attach-shadow.js]: done

**Pull request merged**